From 188522516f7fea434d3ef5b7ff65fe6c1342b9aa Mon Sep 17 00:00:00 2001
Message-Id: <188522516f7fea434d3ef5b7ff65fe6c1342b9aa.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Sun, 18 Jan 2015 01:00:11 +0530
Subject: [PATCH 25/29] MUST_REBASE [VPG]: video/adf/intel: Changes for
 enabling MIPI DSI on Port C

In hardware, DSI Port A is linked with Pipe A and DSI Port C is linked
with Pipe B. DSI Port no is retreived from VBT, based on that pipe is used.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-4868
Change-Id: I39318d2324eaf1ae2010f752bf72176dbb47d21b
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 .../video/adf/intel/core/common/dsi/dsi_config.c   |    1 +
 drivers/video/adf/intel/core/vlv/vlv_dc_config.c   |   12 ++++++++++--
 .../video/adf/intel/include/core/intel_dc_config.h |    5 +++++
 3 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/dsi_config.c b/drivers/video/adf/intel/core/common/dsi/dsi_config.c
index 6907336..6130acf 100644
--- a/drivers/video/adf/intel/core/common/dsi/dsi_config.c
+++ b/drivers/video/adf/intel/core/common/dsi/dsi_config.c
@@ -220,6 +220,7 @@ int dsi_config_init(struct dsi_config *config,
 
 	memcpy(&config->perferred_mode, &mode, sizeof(mode));
 
+	config->pipe = idx;
 	config->changed = 0;
 	config->drv_ic_inited = 0;
 	config->bpp = pi.bpp;
diff --git a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
index 2c3f551..c1be73c 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
@@ -134,7 +134,7 @@ static int vlv_initialize_port(struct vlv_dc_config *vlv_config,
 }
 
 static int vlv_initialize_disp(struct vlv_dc_config *vlv_config,
-			int pipe, enum intel_pipe_type type, u8 disp_no)
+			enum pipe pipe, enum intel_pipe_type type, u8 disp_no)
 {
 	struct vlv_pri_plane *pplane;
 	struct vlv_sp_plane *splane;
@@ -228,6 +228,8 @@ struct intel_dc_config *vlv_get_dc_config(struct pci_dev *pdev, u32 id)
 	struct intel_dc_memory *memory;
 	int err;
 	u8 display_no = 0;
+	u16 port;
+	enum pipe pipe = PIPE_A;
 
 	if (!pdev)
 		return ERR_PTR(-EINVAL);
@@ -262,8 +264,14 @@ struct intel_dc_config *vlv_get_dc_config(struct pci_dev *pdev, u32 id)
 	 * TODO: add dpms config over here and register with adf using
 	 * intel_dc_config_add_power();
 	 */
+	port = intel_get_dsi_port_frm_vbt();
 
-	vlv_initialize_disp(config, PIPE_A, INTEL_PIPE_DSI, display_no++);
+	if (port == DVO_PORT_DSI_A)
+		pipe = PIPE_A;
+	else if (port == DVO_PORT_DSI_C)
+		pipe = PIPE_B;
+
+	vlv_initialize_disp(config, pipe, INTEL_PIPE_DSI, display_no++);
 
 
 	return &config->base;
diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index 9f459cf..e1d4469 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -30,6 +30,11 @@
 
 #define INTEL_ADF_MAX_OVERLAY_ENG_PER_INTF	3
 
+#define DVO_PORT_DSI_A			21
+#define DVO_PORT_DSI_B			22
+#define DVO_PORT_DSI_C			23
+#define DVO_PORT_DSI_D			24
+
 struct intel_dc_component;
 struct intel_plane;
 struct intel_pipe;
-- 
1.7.9.5

